### PR TITLE
Maybe stop identity test flakiness

### PR DESCRIPTION
**Pull request by jamieparkinson**
I _think_ this fixes the issues with these tests (and it re-enables the ones we were skipping). They appear to have been related to asynchronous interactions with the DOM via `@testing-library/react`, and in some places I think could be regarded as "real" bugs, but on the other hand I couldn't confidently tell you why or indeed if this really works.

I did also add a test for the rendering of the user's requested items, and made the app fetch that list in the same way as it does everything else.

### CODE REVIEW

File: identity/webapp/pages/account/registration.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { useEffect, FormEvent, useState } from 'react';
+import { useEffect, FormEvent } from 'react';
 import usePasswordRules from '../../src/frontend/hooks/usePasswordRules';
 import { NextPage } from 'next';
-import { useForm, Controller } from 'react-hook-form';
+import { useForm, Controller, useWatch } from 'react-hook-form';
```
Comment (davidpmccormick) [inline identity/webapp/pages/account/registration.tsx:4]: neat-oh

**Review by davidpmccormick — APPROVED** [review]

File: identity/webapp/src/frontend/MyAccount/ChangePassword.test.tsx [hunk 2]
```diff
@@ -242,11 +242,11 @@ describe('ChangePassword', () => {
     });
   });
 
-  xdescribe('shows an error after submission', () => {
+  describe('shows an error after submission', () => {
     it('when the current password is incorrect', async () => {
       server.use(
         rest.put('/account/api/users/me/password', (req, res, ctx) => {
-          return res(ctx.status(401));
+          return res.once(ctx.status(401));
```
Comment (davidpmccormick) [inline identity/webapp/src/frontend/MyAccount/ChangePassword.test.tsx:249]: Any idea why this is better?

**Review by davidpmccormick — APPROVED** [review]
🚢

File: identity/webapp/src/frontend/MyAccount/ChangePassword.test.tsx (hunk 2 above)
Comment (jamieparkinson) [reply identity/webapp/src/frontend/MyAccount/ChangePassword.test.tsx:249]: I suspect it was something else that fixed things but the intention is that the handler is for that test only (doesn't have to be overridden)

**Pull request merged**